servers/src/main/java/tachyon/master/permission/FileSystemPermissionChecker.java

```diff
@@ -49,7 +50,63 @@ public static synchronized void initializeFileSystem(boolean permissionCheckEnab
   }
 
   /**
-   * This method checks access permission.
+   * Checks requested permission on the last node and basic permission on the whole path.
```

Comment:
Why do we expose the following public API:

checkSelfPermission(user, groups, action, inodes)
checkParentOrAncestorPermission(user, groups, action, inodes)
checkOwner(user, groups, action, inodes)

as opposed to:

checkPermission(user, groups, action, path)